lib/Transport.js

```diff
@@ -247,10 +250,11 @@ class Transport {
 
       const contentEncoding = (result.headers['content-encoding'] || '').toLowerCase()
       const isCompressed = contentEncoding.indexOf('gzip') > -1 || contentEncoding.indexOf('deflate') > -1
-
       /* istanbul ignore else */
       if (result.headers['content-length'] !== undefined) {
         const contentLength = Number(result.headers['content-length'])
+        const heapUsed = process.memoryUsage().heapUsed
+        // nodeJS data type limit check
```

Comment:
move this down to the `else if` bock so that we don't need to get memory usage when circuit break is disabled

Reply:
make sense, will update. Since i used this variable in else if check expression. I might need create a nested if else statements here, e.g.:
```
else if (this.memoryCircuitBreakerEnabled) {
      const heapUsed = process.memoryUsage().heapUsed
      if (contentLength + heapUsed > ....) {...}
}
```

Reply:
make sense, will update. Since i used this variable in else if check expression. I might need create a nested if else statements here, e.g.:
```
else if (this.memoryCircuitBreakerEnabled) {
      const heapUsed = process.memoryUsage().heapUsed
      if (contentLength + heapUsed > ....) {...}
}
```

Reply:
sounds good, and please create functions to improve the readability when necessary